Libraries/graphics/Headers/graphics/PixelFormat.h

```diff
@@ -0,0 +1,149 @@
+/**
+ Copyright (c) 2015-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#ifndef __graphics_PixelFormat_h
+#define __graphics_PixelFormat_h
+
+#include <vector>
+
+#include <cstddef>
+#include <cstdint>
+
+namespace graphics {
+
+/*
+ * Describes the format of pixels in a bitmap. Assumes 8 bits per channel.
+ */
+class PixelFormat {
+public:
+    /*
+     * The channels stored in the bitmap.
+     */
+    enum class Color {
+        /*
+         * Only a gray channel.
+         */
+        Grayscale,
+        /*
+         * RGB color.
+         */
+        RGB,
+    };
+
+    /*
+     * The order of the channels.
+     */
+    enum class Order {
+        /*
+         * Forward order in memory, i.e. big-endian.
+         */
+        Forward,
+        /*
+         * Reversed order in memory, i.e. little-endian.
+         */
+        Reversed,
```

Comment:
That could end up with consistency issues, though, if the fields conflicted.

What's the concern with specifying the format as the combination of different axes?